public/app/features/library-panels/components/LibraryPanelsSearch/LibraryPanelsSearch.tsx

```diff
@@ -29,32 +32,45 @@ export const LibraryPanelsSearch = ({
   variant = LibraryPanelsSearchVariant.Spacious,
   currentPanelId,
   perPage = DEFAULT_PER_PAGE_PAGINATION,
-  showFilter = false,
+  showPanelFilter = false,
+  showFolderFilter = false,
   showSort = false,
   showSecondaryActions = false,
 }: LibraryPanelsSearchProps): JSX.Element => {
   const [searchQuery, setSearchQuery] = useState('');
   const [sortDirection, setSortDirection] = useState<string | undefined>(undefined);
   const [panelFilter, setPanelFilter] = useState<string[]>([]);
+  const [folderFilter, setFolderFilter] = useState<string[]>([]);
   const styles = useStyles2(getStyles);
   const onSortChange = useCallback((sort: SelectableValue<string>) => setSortDirection(sort.value), []);
   const onFilterChange = useCallback((plugins: PanelPluginMeta[]) => setPanelFilter(plugins.map((p) => p.id)), []);
+  const onFolderFilterChange = useCallback(
+    (folders: FolderInfo[]) => setFolderFilter(folders.map((f) => String(f.id!))),
+    []
+  );
```

Comment:
The state is getting pretty complex here, I wonder if it would be beneficial to move this to a reducer via `useReducer` hook?

Reply:
Yes, good suggestion. Will fix!

Reply:
Added RTL tests and reducer.